source/bson-corpus/tests/binary.json

```diff
@@ -82,5 +86,19 @@
             "description": "subtype 0x02 length negative one",
             "bson": "130000000578000600000002FFFFFFFFFFFF00"
         }
+    ],
+    "parseErrors": [
+        {
+            "description": "$uuid missing hyphens",
```

Comment:
I like this test, and think we should keep it, but seems out of line with the current MAY support strings without hyphens that you added to the spec.